lib/deimos/utils/db_poller/base.rb

```diff
@@ -21,17 +21,39 @@ class Base
         # @return [Hash]
         attr_reader :config
 
+        # Method to define producers if a single poller needs to publish to multiple topics.
+        # Producer classes should be constantized
+        # @return [Array<Producer>]
+        def self.multi_producers
+          []
+        end
+
         # @param config [FigTree::ConfigStruct]
         def initialize(config)
           @config = config
           @id = SecureRandom.hex
+          @producers = []
           begin
-            @producer = @config.producer_class.constantize
-          rescue NameError
-            raise "Class #{@config.producer_class} not found!"
-          end
-          unless @producer < Deimos::ActiveRecordProducer
-            raise "Class #{@producer.class.name} is not an ActiveRecordProducer!"
+            @poller_class = @config.poller_class.present? ? @config.poller_class.constantize : nil
+            if self.class.multi_producers.empty?
+              if @config.producer_class.nil?
+                raise 'No producers have been set for this DB poller!'
+              end
+
+              @producer = @config.producer_class.constantize
```

Comment:
There's no need for this instance variable any more. You're going to want to validate all the producers that have been configured.